nicsdru_nidirect_theme.theme

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @file
+ * Functions to support theming in the nicsdru_nidirect_theme theme.
+ */
+
+use Drupal\user\Entity\User;
+
+/**
+ * Implements hook_preprocess_field().
+ */
+function nicsdru_nidirect_theme_preprocess_field(&$variables) {
+  // Implement audit link.
+  if (isset($variables['element'])
+    && isset($variables['element']['#entity_type'])
+    && ($variables['element']['#entity_type'] == 'node')
+  ) {
+    // Only start this processing if the 'nidirect_workflow'
+    // module is installed.
+    if (\Drupal::service('module_handler')->moduleExists('nidirect_workflow')) {
+      $content_type = $variables['element']['#bundle'];
+      switch ($content_type) {
+        case 'health_condition':
+          // For health conditions, just send the user to the node edit
+          // form, where they can set the 'next review date' to a date
+          // in the future to remove this node from the 'needs audit' view.
+          $node = \Drupal::routeMatch()->getParameter('node');
+          if (!empty($node) && is_object($node)) {
+            $nid = $node->id();
+            if (!empty($nid)) {
+              // See if audit is due.
+              $next_review_date = $node->get('field_next_review_date')->value;
+              if (strtotime($next_review_date) < strtotime("now")) {
+                // Next review date is in the past,
+                // so this node is due for audit - display the audit link
+                // if the user is allowed to see it.
+                $account = User::load(\Drupal::currentUser()->id());
+                if ($account->hasPermission('audit content')) {
```

Comment:
Eyeballing this, the cyclomatic complexity here is probably in the region of moderately complex as per PHPMD's guideline: https://phpmd.org/rules/codesize.html

You can avoid this by either breaking the code into smaller functions with discrete return types/purposes as well as using the early return pattern to exit the function early on using the broadest conditions for not needing to proceeed any further.